src/functional/program_processors/codegens/gtfn/gtfn_module.py

```diff
@@ -43,33 +47,118 @@ class GTFNTranslationStep(
     language_settings: languages.LanguageWithHeaderFilesSettings = cpp_interface.CPP_DEFAULT
     enable_itir_transforms: bool = True  # TODO replace by more general mechanism, see https://github.com/GridTools/gt4py/issues/1135
 
+    def _process_arguments(
+        self,
+        program: itir.FencilDefinition,
+        args: tuple[Any, ...],
+        offset_provider: dict[str, Connectivity],
+    ):
+        """Given an ITIR program and arguments generate the interface parameters and arguments."""
+        parameters: list[
+            interface.ScalarParameter | interface.BufferParameter | interface.ConnectivityParameter
+        ] = []
+        parameter_args: list[str] = ["gridtools::fn::backend::naive{}"]
+        connectivity_args: list[str] = []
+
+        # handle parameters and arguments of the program
+        # TODO(tehrengruber): The backend expects all arguments to a stencil closure to be a SID
+        #  so transform all scalar arguments that are used in a closure into one before we pass
+        #  them to the generated source. This is not a very clean solution and will fail when
+        #  the respective parameter is used elsewhere, e.g. in a domain construction, as it is
+        #  expected to be scalar there (instead of a SID). We could solve this by:
+        #   1.) Extending the backend to support scalar arguments in a closure (as in embedded
+        #       backend).
+        #   2.) Use SIDs for all arguments and deref when a scalar is required.
+        closure_scalar_parameters = (
+            eve.trees.pre_walk_values(
+                eve.utils.XIterable(program.closures).getattr("inputs").to_list()
+            )
+            .if_isinstance(itir.SymRef)
+            .getattr("id")
+            .map(str)
+            .to_list()
+        )
+        for obj, program_param in zip(args, program.params):
+            # parameter
+            parameter = get_param_description(program_param.id, obj)
+            parameters.append(parameter)
+
+            # argument expression
+            if (
+                isinstance(parameter, interface.ScalarParameter)
+                and parameter.name in closure_scalar_parameters
+            ):
+                # convert into sid
+                parameter_args.append(f"gridtools::stencil::global_parameter({parameter.name})")
+            else:
+                # pass as is
+                parameter_args.append(parameter.name)
```

Comment:
I propose to do one small step more and put this into a function. This is a big improvement in my opinion, because, I can clearly see that in the next 15 lines
- only `args` and `program.params` is used
- plus an additional weird thing that we should get rid off